### PR TITLE
Add test cases for version bump

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR adds 30+ normal and 50+ edge test cases for version bump action

### CODE REVIEW

File: packages/version-bump/src/main.ts [hunk 1]
```diff
@@ -35,10 +35,17 @@ export async function run(): Promise<void> {
 			throw new Error('Invalid version! Version does not match the pattern');
 		}
 
+		// remove empty matches from groups to avoid them overriding defaults
+		const nonEmptyVersionParts = filter(Boolean, versionPartsMatch?.groups);
```
Comment (nicolad) [inline packages/version-bump/src/main.ts:39]: :+1:

File: packages/version-bump/src/main.ts [hunk 2]
```diff
@@ -35,10 +35,17 @@ export async function run(): Promise<void> {
...
 		// build version parts by setting defaults
-		const versionParts = { ...DEFAULT_VERSION_PARTS, ...versionPartsMatch?.groups };
+		const versionParts = { ...DEFAULT_VERSION_PARTS, ...nonEmptyVersionParts };
 
-		let { major, minor, patch, release, build } = versionParts;
+		// extract `release` from the parts as it's the only non-numeric part
+		let { release } = versionParts;
+
+		// make sure the numeric parts of the version are numbers
+		let { major, minor, patch, build } = map(Number, versionParts);
```
Comment (nicolad) [inline packages/version-bump/src/main.ts:48]: `map(Number, versionParts);` : :heart:

File: packages/version-bump/src/main.ts [hunk 3]
```diff
@@ -62,6 +69,22 @@ export async function run(): Promise<void> {
...
+						major--;
+						minor = 9;
+						patch = 9;
+					} else {
+						minor--;
+					}
+				} else {
+					patch--;
+				}
+				release = releaseTypes.release;
```
Comment (nicolad) [inline packages/version-bump/src/main.ts:85]: `releaseTypes.release` reads a bit odd, some internals might be improved

File: packages/version-bump/src/tests/main.test.ts [hunk 4]
```diff
@@ -53,18 +45,38 @@ describe('version bump main tests', () => {
 		jest.restoreAllMocks();
 	});
 
-	it('returns the given input', () => {
-		for (const { type, inputVer, outputVer } of testCases) {
+	it('returns the expected output for a given input', async () => {
+		checkForDuplicateCases(mainNormalTestCases);
```
Comment (nicolad) [inline packages/version-bump/src/tests/main.test.ts:49]: :+1:

File: packages/version-bump/src/tests/main.test.ts [hunk 5]
```diff
@@ -53,18 +45,38 @@ describe('version bump main tests', () => {
...
+
+			await expect(run()).resolves.not.toThrowError();
+
 			expect(output['new-version']).toBe(outputVer);
 			contentMock.mockRestore();
 		}
 	});
+
+	it('errs for an invalid input', async () => {
+		for (const edgeCase of [null, undefined, '', '0', '0.0', '1', '1.0']) {
```
Comment (nicolad) [inline packages/version-bump/src/tests/main.test.ts:67]: nice use of for of :taco:

File: packages/version-bump/src/tests/utils.ts [hunk 6]
```diff
@@ -20,8 +22,213 @@ export const getMockedFileContents = (path: string, version?: string): string =>
...
             License: GPL2`;
 	}
 	return '';
 };
+
+type MainTestCase = {
+	type: BumpType;
+	inputVer: string;
+	outputVer?: string;
+};
```
Comment (nicolad) [inline packages/version-bump/src/tests/utils.ts:35]: :clap: :clap: :clap:

**Review by nicolad — APPROVED** [review]
image

File: packages/version-bump/src/main.ts (hunk 3 above)
Comment (manzoorwanijk) [reply packages/version-bump/src/main.ts:85]: Yeah, it should be something like `releaseTypes.production`, but it's some thing that's already used by many addons, so I didn't change the name.

**Pull request merged**